### PR TITLE
[Feature] Move user group permission to PG db

### PR DESCRIPTION
**Pull request by roycephan**
#### Status

READY

#### What's this PR do?

- [x] add crud user, group service
- [x] add group service
- [x] update auto migrate
- [x] move user to pg
- [x] authenticate by group
- [x] update test
- [x] update travis
- [x] allow admin update permission

### CODE REVIEW

File: backend/config/database/pg/pg.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+		return nil, nil, err
+	}
+
+	return db, func() {
+		db.Close()
+	}, nil
+}
+
+// MigrateTables .
+func MigrateTables(db *gorm.DB) error {
```
Comment (hieunmce) [inline backend/config/database/pg/pg.go]: THis function should have better name (seed create table etc..)

File: backend/config/database/pg/pg.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+	}
+
+	if err := db.Model(&domain.UserGroup{}).AddForeignKey("group_id", "groups(id)", "RESTRICT", "RESTRICT").Error; err != nil {
+		return err
+	}
+	if err := db.Model(&domain.UserGroup{}).AddForeignKey("user_id", "users(id)", "RESTRICT", "RESTRICT").Error; err != nil {
+		return err
+	}
+
+	group1ID := domain.MustGetUUIDFromString("18a5cac3-1024-43c8-acbe-22be76006040")
```
Comment (hieunmce) [inline backend/config/database/pg/pg.go]: there ids should be in constants (ex: SeedGroup1ID)

File: backend/config/database/pg/pg.go [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+	userID := domain.MustGetUUIDFromString("18a5cac3-1024-43c8-acbe-22be76006041")
+	permission1ID := domain.MustGetUUIDFromString("18a5cac3-1024-43c8-acbe-22be76006042")
+	permission2ID := domain.MustGetUUIDFromString("18a5cac3-1024-43c8-acbe-22be76006043")
+	group1 := domain.Group{
+		Model:       domain.Model{ID: group1ID},
```
Comment (hieunmce) [inline backend/config/database/pg/pg.go]: all of this too

**Review by hieunmce — CHANGES_REQUESTED** [review]

File: backend/config/database/pg/pg.go (hunk 2 above)
Comment (phanletrunghieu) [reply backend/config/database/pg/pg.go]: Due to the result returned from `MustGetUUIDFromString` function, I can't assign it to `group1ID` by `const`

**Review by hieunmce — APPROVED** [review]

**Review by hieunmce — APPROVED** [review]

**Pull request merged**